lnd.go

```diff
@@ -667,56 +667,52 @@ func Main(cfg *Config, lisCfg ListenerCfg, shutdownChan <-chan struct{}) error {
 	}
 	defer rpcServer.Stop()
 
-	// If we're not in regtest or simnet mode, We'll wait until we're fully
-	// synced to continue the start up of the remainder of the daemon. This
-	// ensures that we don't accept any possibly invalid state transitions, or
-	// accept channels with spent funds.
-	if !(cfg.Bitcoin.RegTest || cfg.Bitcoin.SimNet ||
```

Comment:
It's quite possible that this is causing some of the flakes. Though this condition was added specifically for regtest and simnet networks where it can happen that no block is produced for more than 2 hours. In that case lnd decides that it is _not_ synced to chain, even if it's at the chain tip. Maybe we just need to introduce a new boolean variable that is set to true for integration tests so we always execute this branch in itests?

Reply:
> Maybe we just need to introduce a new boolean variable that is set to true for integration tests so we always execute this branch in itests?

If we always want to execute this branch, then maybe we don't need to have the boolean variable? Meanwhile, if the 2 hours setting was not specified in the BOLTs, could it be set to be configurable?

Reply:
What I meant is: We want the old behavior on normal regtest setups. We want `lnd` to continue and not block if there was no block for 2 hours or longer. But in the integration tests we never run into the situation where we have 2 hour old blocks and therefore always want to execute the branch and block on sync.

I was mistaken, I think the 2 hours come from `bitcoind` itself, it thinks it's out of sync with the network if the last block is older than 2 hours. So I think that's nothing we can fix in `lnd` itself.

Reply:
I see. So when 2 hours passed and no blocks produced, `lnd` with the `bitcoind` backend will get stuck here. If that's the case, maybe it's better to let it stuck (concerning safety)? Meanwhile, this PR doesn't change the old behavior for mainnet/testnet, it just adds the missing part on checking sync state in the integration test.